crates/polars-plan/src/dsl/options.rs

```diff
@@ -298,6 +298,18 @@ pub enum SinkType {
         file_type: FileType,
         cloud_options: Option<polars_io::cloud::CloudOptions>,
     },
+    Partition {
+        path_f_string: Arc<PathBuf>,
+        file_type: FileType,
+        variant: PartitionVariant,
+        cloud_options: Option<polars_io::cloud::CloudOptions>,
+    },
+}
+
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum PartitionVariant {
+    MaxSize(IdxSize),
```

Comment:
Will this add 

- by keys 
- Round Robin
- others?

Reply:
Yes, that is the idea. I basically want a generic interface to the partitioning schemes so that we don't need new infrastructure for every scheme.